## WEEK 1 PRACTICE

Question 1

Which of the following options are the class names in the iris dataset?

(a) ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']

(b) ['setosa' 'versicolor' 'virginica']

(c) ['setosa' 'virginica']

(d) ['sepal length (cm)', 'petal length (cm)', 'petal width (cm)']

Correct Answer: Option 2
(1 marks)

In [5]:
from sklearn.datasets import load_iris
data = load_iris()
print(data.feature_names)
print(data.target_names)

['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']
['setosa' 'versicolor' 'virginica']


Question 2

Generate 200 samples with 10 features for multiple regression problem with 10 outputs and find the shapes of the feature matrix and the label vector.

(a) Feature matrix shape: (200,10), label vector shape: (200,10)

(b) Feature matrix shape: (10,200), label vector shape: (10,200)

(c) Feature matrix shape: (200,200), label vector shape: (10,10)

(d) Feature matrix shape: (10,10), label vector shape: (200,200)

Correct Answer: Option 1

(1 Marks)

In [6]:
from sklearn.datasets import make_regression
# ?make_regression

x,y = make_regression(n_samples=200, n_features=10, n_targets=10, shuffle=True)
print(x.shape)
print(y.shape)

(200, 10)
(200, 10)


Question 3

Find the shapes of the feature matrix and the label vector for a multilabel classification problem with 200 samples, 10 features, 7 labels and on an average 3 labels per example.

(a) Feature matrix: (200,10), Label vector: (200,7)

(b) Feature matrix: (200,200), Label vector: (10,7)

(c) Feature matrix: (200,7), Label vector: (200,10)

(d)Feature matrix: (199,9), Label vector: (199,6)

Correct Answer: Option 1

(1 Marks)

In [7]:
from sklearn.datasets import make_multilabel_classification
x,y = make_multilabel_classification(n_samples=200, n_features=10, n_classes=7, n_labels=3)
print(x.shape)
print(y.shape)
print(x[:10])
print(y[:10])

(200, 10)
(200, 7)
[[ 3.  2.  5.  8.  4.  4.  2.  3.  4. 10.]
 [ 8.  9.  5.  6.  6.  4.  2.  2.  3.  6.]
 [ 6.  1.  4.  7.  6.  3.  2.  6.  7.  4.]
 [ 2.  4.  6.  5.  6.  9.  2.  4.  6.  5.]
 [ 8.  1.  0. 13.  4.  2.  1.  3.  7.  8.]
 [ 3.  9.  4.  2.  3.  7.  3.  5.  1.  7.]
 [11.  4.  0.  8.  5.  2.  5.  2.  6.  6.]
 [ 6.  1.  6.  6.  5.  5.  0.  5.  5.  8.]
 [17.  3.  6.  0.  9.  5.  3.  4. 10. 10.]
 [ 2.  5.  7. 17.  7.  0.  0. 10.  7.  1.]]
[[0 0 0 1 1 0 1]
 [0 0 1 0 1 0 0]
 [0 1 1 1 0 0 1]
 [0 0 1 0 1 0 0]
 [0 0 0 1 0 0 0]
 [1 0 0 0 1 0 0]
 [0 0 1 1 0 0 0]
 [1 1 1 1 1 0 1]
 [1 0 1 0 0 0 0]
 [0 1 0 0 0 0 0]]


Question 4

Which of the following functions can be used to load the diabetes dataset as a dataframe?

(a) x,y= load_diabetes(as_frame=True,return_X_y=True)

(b)x,y= load_diabetes(as_frame=True,return_X_y=False)

(c) print(load('diabetes'))

(d) load diabetes_frame

Correct Answer: Option 1

(1 Mark)

Question 5

What are the datatypes of the feature matrix and the target matrix in the diabetes dataset (by using load_diabetes() from sklearn)?

(a) DataFrame and Series

(b) Series and DataFrame

(c) Both are Series

(d) Both are DataFrames

Correct Answer: Option 1

(1 Marks)

In [8]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
from sklearn.datasets import load_diabetes
x,y = load_diabetes(as_frame=True, return_X_y=True) # x is the feature matrix, y is target matrix
# return_X_ybool, default=False, If True, returns (data,target) instead of a bunch object. See below for more information about the data and target object.
data = x+ y

In [9]:
print(type(x)) # PRINT THE DATATYPE OF FEATURE MATRIX
print(type(y)) # print the datatype of the target matrix
print(type(data))

<class 'pandas.core.frame.DataFrame'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.frame.DataFrame'>


Question 6: Which of the following options can be the output of the following code?

In [ ]:
#codeblock
from sklearn.datasets import make_blobs
X, y = make_blobs(n_samples=10, centers=3, n_features=2, random_state=42)
print ("Feature matrix shape:", X.shape)

(a) Feature matrix shape: (10, 2)

(b) Feature matrix shape: (2, 10)

(c) Feature matrix shape: (10, 10)

(d) Feature matrix shape: (2, 2)

Correct Answer: Option 1

(2 Marks)

In [13]:
from sklearn.datasets import make_blobs
X,y = make_blobs(n_samples=10, centers=3, n_features=2, random_state=42)
print("Feature matrix shape:", X.shape)
# print("Label shape:", y.shape)


Feature matrix shape: (10, 2)


In [14]:
X,y = make_blobs(n_samples=10, centers=3, n_features=2, random_state=42)
print("Feature matrix shape:", X.shape)
print("Label shape:", y.shape)

Feature matrix shape: (10, 2)
Label shape: (10,)


## WEEK 1 GRADED

Question 1

Find shape of the Wisconsin breast cancer dataset (https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Original)).

(a)(570, 35)

(b) (569, 33)

(c) (33,569)

(d) (35,570)

Correct Answer: Option 2


Question 2

How many benign ( B ) tumour cases are present in the Wisconsin breast cancer dataset (https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Original))?

(NAT Question)

Correct Answer: 357

(2 Marks)

Question 3

How many malignant ( M ) tumour cases are present in the Wisconsin breast cancer dataset (https://archive.ics.uci.edu/ml/datasets/Breast+Cancer+Wisconsin+(Original)).?

(NAT Question)

Correct Answer: 212

(2 Marks)

In [16]:
import pandas as pd
import numpy as np
# import pandas_profiling
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import StandardScaler

In [25]:
from sklearn.datasets import load_breast_cancer
X,y = load_breast_cancer(as_frame=True, return_X_y=True)
X.shape

(569, 30)

In [26]:
y.value_counts()

1    357
0    212
Name: target, dtype: int64

Instructions for Questions 4-6: Use the following code for importing California Housing dataset from sklearn. Also, name the dataset as housing_data and print description of the dataset.



In [27]:
#codeblock
from sklearn.datasets import fetch_california_housing
housing_data = fetch_california_housing()
print(housing_data.DESCR)

.. _california_housing_dataset:

California Housing dataset
--------------------------

**Data Set Characteristics:**

    :Number of Instances: 20640

    :Number of Attributes: 8 numeric, predictive attributes and the target

    :Attribute Information:
        - MedInc        median income in block group
        - HouseAge      median house age in block group
        - AveRooms      average number of rooms per household
        - AveBedrms     average number of bedrooms per household
        - Population    block group population
        - AveOccup      average number of household members
        - Latitude      block group latitude
        - Longitude     block group longitude

    :Missing Attribute Values: None

This dataset was obtained from the StatLib repository.
https://www.dcc.fc.up.pt/~ltorgo/Regression/cal_housing.html

The target variable is the median house value for California districts,
expressed in hundreds of thousands of dollars ($100,000).

This dataset was derived

Question 4: Find out the shape of the feature matrix in california housing dataset.

(a) (20118,19)

(b) (20640, 8)

(c) (8, 8)

(d) (8, 20640)

Correct Answer: Option 2

(1 Marks)

In [28]:
print(housing_data.data.shape)
# print(housing_data.data[:5])

(20640, 8)


Question 5: Find out the labels of the first five attributes in california housing dataset. Note, the labels are real-valued.

(a) [4.526, 3.413, 3.422, 1.22, 6.02]

(b) [4.526, 3.585, 3.521, 3.413, 3.422]

(c) [3.413, 3.422, 4.526, 3.585, 1.249]

(d) [4.526, 502.0, 3.521, 3.413, 1025.5]

Correct Answer: Option 2

(1.5 Marks)

In [29]:
print(housing_data.target.shape)
print(housing_data.target[:5])

(20640,)
[4.526 3.585 3.521 3.413 3.422]


Question 6: Find out the name of the class label in california housing dataset(target matrix)?

(a)['MedHouseVal']

(b) ['MedInc']

(c) ['AveRooms']

(d) ['Population']

Correct Answer: Option 1

(1 Marks)

In [30]:
print(housing_data.feature_names)
print(housing_data.target_names)

['MedInc', 'HouseAge', 'AveRooms', 'AveBedrms', 'Population', 'AveOccup', 'Latitude', 'Longitude']
['MedHouseVal']
